apps/src/netsim/NetSimEntity.js

```diff
@@ -110,3 +110,25 @@ NetSimEntity.prototype.getTable_ = function () {
 NetSimEntity.prototype.buildRow_ = function () {
   return {};
 };
+
+/**
+ * Destroys all provided entities (from remote storage) asynchronously, and
+ * calls onComplete when all entities have been destroyed and/or an error occurs.
+ * @param {NetSimEntity[]} entities
+ * @param {!NodeStyleCallback} onComplete
+ */
+NetSimEntity.destroyEntities = function (entities, onComplete) {
+  if (entities.length === 0) {
+    onComplete(null, true);
+    return;
+  }
+
+  entities[0].destroy(function (err, result) {
+    if (err) {
+      onComplete(err, result);
+      return;
+    }
+
+    NetSimEntity.destroyEntities(entities.slice(1), onComplete);
```

Comment:
That gives me two reasons to leave this the way it is:

1. I'd rather leave the tail of the list un-deleted on error.  Un-deleted messages will be left "on the wire" and picked up again on the next tableChange event.  Leaving any N/10 items on the wire would produce out-of-order routing, which IS something we want to explore, but it should be done intentionally by the simulation, not accidentally as a result of a network error.

2. At some point we're going to animate the messages being sent, a process that will scale up with the number of packets going out, at a higher coefficient than the actual sequential sending is likely to.  We might even want to start each packet send as the animation for the previous send completes.  This model sticks closer to that eventual outcome.